Announcement that the snapshot branches are moving (#925)
rust-lang/blog.rust-lang.org: posts/2022-02-14-crates-io-snapshot-branches.md

---
layout: post
title: "Crates.io Index Snapshot Branches Moving"
author: the crates.io team <https://www.rust-lang.org/governance/teams/crates-io>
release: false
---

Every so often, the [crates.io index](https://github.com/rust-lang/crates.io-index)'s Git history
is [squashed into one
commit](https://internals.rust-lang.org/t/cargos-crate-index-upcoming-squash-into-one-commit/8440)

> to minimize the history Cargo needs to download. When the index is squashed, we have saved snapshot
> branches to preserve the history of crate publishes.
> 
> Those branches are using server resources because of the recompression of objects that are

Reviewer: ```suggestion
to minimize the history Cargo needs to download. When the index is squashed, we save snapshots
to preserve the history of crate publishes.

Currently, those snapshots are stored as branches in the main index Git repository. This uses server resources because of the recompression of objects that are
```

> delta-compressed against objects that are only referenced in branches not fetched by the client. We

Reviewer 2: This sentence is a bit hard to parse. Maybe we can simplify it a bit more?

> Those branches are using server resources though, as the server still has to consider their contents whenever Cargo asks for the `master` branch.

> will be deleting the snapshot branches to ensure that all objects referenced in the master branch

Reviewer: ```suggestion
will be deleting the snapshot branches from this repository to ensure that all objects referenced in the master branch
```

---
layout: post
title: "Crates.io Index Snapshot Branches Moving"

> author: the crates.io team <https://www.rust-lang.org/governance/teams/crates-io>

Reviewer 3: The blog doesn't render markdown for the author field.
This should be just the following:

```suggestion
author: The Crates.io Team
```

(that's used by at least one other blog post)
